### PR TITLE
root: sha256 change on latest versions

### PR DESCRIPTION
**Pull request by wdconinc**
The ROOT 6.30.02 version sha256 added in https://github.com/spack/spack/pull/41289 appears incorrect or changed:
```console
$ curl -L https://root.cern/download/root_v6.30.02.source.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  176M  100  176M    0     0  7529k      0  0:00:23  0:00:23 --:--:-- 11.5M
7965a456d1ad1ee0d5fe4769bf5a8fec291af684ed93db0f3080a9c362435183  -
```

Also ROOT 6.28.10 the other version sha256 changed in https://github.com/spack/spack/pull/41289 is incorrect or changed, but 6.30.00 is fine:
```console
$ curl -L https://root.cern/download/root_v6.28.10.source.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  177M  100  177M    0     0  6403k      0  0:00:28  0:00:28 --:--:-- 9258k
69d6fdeb607e6b20bd02c757fa6217024c0b6132c1e9b1dff4d85d9a2bb7e51e  -
$ curl -L https://root.cern/download/root_v6.30.00.source.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  176M  100  176M    0     0  6570k      0  0:00:27  0:00:27 --:--:-- 6790k
0592c066954cfed42312957c9cb251654456064fe2d8dabdcb8826f1c0099d71  -
```

### CODE REVIEW

**Comment by wdconinc** [comment]
@jmcarcell Can you comment on how you determined those sha256 checksums? It seems strange that you would find different sha256 checksums, so I'm wondering if you are looking at different files. Also, of course, please confirm the ones I'm getting so it's not a problem on my end.

**Comment by wdconinc** [comment]
I think I see what's going on. Could it be that you're using the github tag archives?
```console
$ curl -L https://github.com/root-project/root/archive/refs/tags/v6-30-02.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  176M    0  176M    0     0  8352k      0 --:--:--  0:00:21 --:--:-- 6404k
51a09c86ffa94089abac76daa3adc20812efc6c93b427697b843b12d73e145de  -
```

Those github archives are not the ones used by spack. If a project has other archives they are usually preferred because automatic github archives can change (e.g. https://github.blog/changelog/2023-01-30-git-archive-checksums-may-change/, reverted for now after widespread uproar, but the link says all).

Also, version 6.26.12 is not even on the ROOT server, so that will result in a failed download. I'll remove that version again, and instead add 6.26.14, which is on the ROOT server (https://github.com/root-project/root/compare/v6-26-12...v6-26-14, just formatting diffs).

**Review by jmcarcell — APPROVED** [review]

File: var/spack/repos/builtin/packages/root/package.py [hunk 1]
```diff
@@ -34,14 +34,14 @@ class Root(CMakePackage):
...
+    version("6.30.02", sha256="7965a456d1ad1ee0d5fe4769bf5a8fec291af684ed93db0f3080a9c362435183")
     version("6.30.00", sha256="0592c066954cfed42312957c9cb251654456064fe2d8dabdcb8826f1c0099d71")
-    version("6.28.10", sha256="fc6a2d6c7cba853b0cfd6bd9514c90e9df50e1743899bc1db0472ee6f4e65a0a")
+    version("6.28.10", sha256="69d6fdeb607e6b20bd02c757fa6217024c0b6132c1e9b1dff4d85d9a2bb7e51e")
     version("6.28.06", sha256="af3b673b9aca393a5c9ae1bf86eab2672aaf1841b658c5c6e7a30ab93c586533")
     version("6.28.04", sha256="70f7f86a0cd5e3f2a0befdc59942dd50140d990ab264e8e56c7f17f6bfe9c965")
     version("6.28.02", sha256="6643c07710e68972b00227c68b20b1016fec16f3fba5f44a571fa1ce5bb42faa")
     version("6.28.00", sha256="afa1c5c06d0915411cb9492e474ea9ab12b09961a358e7e559013ed63b5d8084")
-    version("6.26.12", sha256="229daa0749e3e31b6e0dedc58b6838dbfc1180b4aba4741883b617b0b4fbc966")
+    version("6.26.14", sha256="3b891ea23216234b90bf62397b036e9982607b687475f80d378e2f4bcb54a197")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/root/package.py]: I am getting "81045a35a1958906c3eadecd9b01127a6087dadc19385af8e92ae5c84f06ce35" for this version (from 'spack checksum').

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/root/package.py]: This is the only version in the package that I cannot confirm.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Issue with only one of the version sha256s.

File: var/spack/repos/builtin/packages/root/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/root/package.py]: Yeah, not sure why I ended up with the wrong hash (how can hashes be so hard??). I'm now also getting:
```suggestion
    version("6.26.14", sha256="81045a35a1958906c3eadecd9b01127a6087dadc19385af8e92ae5c84f06ce35")
```

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**